src/main/java/org/folio/qm/service/impl/LinksServiceImpl.java

```diff
@@ -58,7 +56,7 @@ private InstanceLinks extractLinks(QuickMarc quickMarc) {
   private void populateLinks(QuickMarc qmRecord, InstanceLinks instanceLinks) {
     instanceLinks.getLinks().forEach(instanceLink -> {
       var fields = qmRecord.getFields().stream()
-        .filter(fieldItem -> instanceLink.getBibRecordTag().equals(fieldItem.getTag()))
+        .filter(fieldItem -> instanceLink.getAuthorityId().equals(fieldItem.getAuthorityId()))
```

Comment:
Filtering by authorityId is present further. It'd reasonable to have 'linkId' filtering here